mm2src/coins/eth.rs

```diff
@@ -3430,19 +3430,23 @@ impl EthCoin {
     }
 
     pub async fn get_tokens_balance_list(&self) -> Result<HashMap<String, CoinBalance>, MmError<BalanceError>> {
-        let coin = self.clone();
-        let mut token_balances = HashMap::new();
-        for (token_ticker, info) in self.get_erc_tokens_infos().iter() {
-            let balance_as_u256 = coin.get_token_balance_by_address(info.token_address).await?;
-            let balance_as_big_decimal = u256_to_big_decimal(balance_as_u256, info.decimals)?;
-            let balance = CoinBalance {
-                spendable: balance_as_big_decimal,
-                unspendable: BigDecimal::from(0),
+        let selfi = self.clone();
+        let mut requests = Vec::new();
+        for (token_ticker, info) in self.get_erc_tokens_infos() {
+            let coin = selfi.clone();
```

Comment:
Is it cloned to be cloned one more time inside get_token_balance_by_address?

Reply:
If we remove the first clone we will get a clippy warning. I will capture `self` by reference using a closure instead.

Reply:
done